src/zenml/integrations/whylogs/steps/whylogs_profiler.py

```diff
@@ -0,0 +1,129 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import datetime
+from typing import Dict, Optional, cast
+
+import pandas as pd
+from whylogs import DatasetProfile  # type: ignore
+
+from zenml.integrations.whylogs.whylogs_utils import WhylogsContext
+from zenml.steps.step_interfaces.base_analyzer_step import (
+    BaseAnalyzerConfig,
+    BaseAnalyzerStep,
+)
+from zenml.steps.utils import (
+    INSTANCE_CONFIGURATION,
+    PARAM_CREATED_BY_FUNCTIONAL_API,
+    PARAM_ENABLE_CACHE,
+)
+
+
+class WhylogsProfilerConfig(BaseAnalyzerConfig):
+    """Config class for the WhylogsProfiler step.
+
+    The configuration fields are:
+
+        dataset_name: the name of the dataset (Optional). If not specified,
+            the pipeline step name is used
+        dataset_timestamp: timestamp to associate with the generated
+            dataset profile (Optional). The current time is used if not
+            supplied.
+        tags: custom metadata tags associated with the whylogs profile
+
+    Also see `WhylogsContext.log_dataframe`.
+
+    """
+
+    dataset_name: Optional[str] = None
+    dataset_timestamp: Optional[datetime.datetime]
+    tags: Optional[Dict[str, str]] = None
+
+
+class WhylogsProfilerStep(BaseAnalyzerStep):
+    """Simple step implementation which generates a whylogs data profile from a
+    a given pd.DataFrame"""
+
+    def entrypoint(  # type: ignore[override]
+        self,
+        dataset: pd.DataFrame,
+        config: WhylogsProfilerConfig,
+        context: WhylogsContext,
+    ) -> DatasetProfile:
+        """Main entrypoint function for the whylogs profiler
+
+        Args:
+            dataset: pd.DataFrame, the given dataset
+            config: the configuration of the step
+        Returns:
+            whylogs profile with statistics generated for the input dataset
+        """
+        profile = context.log_dataframe(
+            dataset, dataset_name=config.dataset_name, tags=config.tags
+        )
+        return profile
+
+
+def whylogs_profiler_step(
```

Comment:
So this utility is simply to create a new instance of the step to get around that error you ran into with duplicate steps?

Reply:
Yes.